test/nodes/test_document_merger.py

```diff
@@ -0,0 +1,97 @@
+from haystack import Document
+from haystack.nodes.other.document_merger import DocumentMerger
+
+doc_dicts = [
+    {
+        "meta": {
+            "name": "name_1",
+            "year": "2020",
+            "month": "01",
+            "flat_field": 1,
+            "nested_field": {1: 2, "a": 5, "c": {"3": 3}},
```

Comment:
I see only one small case left, which is the case of additional nested keys.

```suggestion
            "nested_field": {1: 2, "a": 5, "c": {"3": 3}, "d": "I will be dropped by the meta merge algorithm"},
```